TAO/orbsvcs/FTRT_Event_Service/Factory_Service/EventChannelFactory_i.cpp

```diff
@@ -124,7 +124,7 @@ CORBA::Object_ptr EventChannelFactory_i::create_process (
 
   ORBSVCS_DEBUG((LM_DEBUG, "Command Line : %s\n", str.c_str()));
 
-  options.command_line(str.c_str());
+  options.command_line(ACE_TEXT("%s"), str.c_str());
```

Comment:
I'm not sure this is correct for wchar builds.
ACE_Process_Options will use `ACE_OS::vsnprintf(wchar_t*, size_t, const wchar_t*, va_list)` which is just `vswprintf` from C99/POSIX.  `vswprintf` can insert a `const wchar_t*` arg but uses the `%ls` formatter for that, not `%s`.  The `%s` formatter (although encoded within a wide string) tells it to read a `const char*` from the va_list.
But I could be missing something...

Reply:
Yes, this will not work, there is a macro for that as far as I remembe

Reply:
I've added commit c320b80 which conditionally uses ACE_TEXT("%ls") instead of ACE_TEXT("%s"), using similar patterns seen elsewhere (including in ACE/ace/Process.cpp).

Reply:
I considered adding overloaded member functions for `ACE_Process_Options::setenv()` and `ACE_Process_Options::command_line()`, but these would require different names because of ambiguous name resolution issues. (A C++11 workaround using the technique in https://stackoverflow.com/a/30761816 is most likely an ABI change, which I suspect is not desirable).

I have experimented with adding methods to ACE_Process_Options:
```
int setenv_value (const ACE_TCHAR *variable_name, const ACE_TCHAR *value);
int command_line_str (const ACE_TCHAR * args);
```
and adapting various callers of `setenv()` and `command_line()` appropriately.

I'm not sure that this is the best approach. Happy to consider different methods based on preferences of experienced ACE contributors.

Reply:
I've tried a different approach instead of the overloaded member functions suggested above.

I've defined `ACE_TEXT_PRIs` to `"s"` or `"ls"`, depending upon the WCHAR mode. This similar to PRIxxx macros in <inttypes.h> (e.g., PRIu64). This does not contain the leading `%`, to allow the caller to use flags or width specifiers as required.

I've then replaced the existing -Wformat-security fixes appropriately.

For example, instead of:
```
options.command_line(ACE_TEXT("%s"), str.c_str());
```
it's now
```
options.command_line(ACE_TEXT("%") ACE_TEXT_PRIs, str.c_str());
```

This is in commit f5e5d7759a9 in my pull request.

Reply:
I agree that having the PRIs macro available is better than not having it.  One complication of this is that it can't be used for format strings that are passed to `ACE_Log_Msg::log()`.  But I guess that's not directly relevant here.

But I suggest we also consider providing methods on these classes that accept strings (not format strings, plain strings) of either narrow or wide characters.  That's much easier to use correctly and it lets the type system enforce correctness instead of going through varargs.

Reply:
@mitza-oci : Do you have suggestions on method names for direct string (non-format-string) methods based on existing ACE conventions? I don't have strong opinions on this, versus the names I experimented with above (`setenv_value()` and `command_line_str()`)

Reply:
I'm OK with the names you've listed.  I don't think ACE has a convention for it.